vss-processor-plugin/src/main/java/org/eclipse/kuksa/vssprocessor/plugin/VssProcessorPlugin.kt

```diff
@@ -0,0 +1,132 @@
+ * KSP VSS Processor. This is necessary because the Symbol Processor does not have access to the android assets folder.
+ */
+class VssProcessorPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        val extension = project.extensions.create<VssProcessorPluginExtension>(EXTENSION_NAME)
+
+        // The extension variables are only available after the project has been evaluated
+        project.afterEvaluate {
+            val buildDir = layout.buildDirectory.asFile.get()
+            val buildDirPath = buildDir.absolutePath
+            val vssDir = "${rootDir}${fileSeparator}$VSS_FOLDER_NAME"
+
+            val provideVssDefinitionTask =
+                project.tasks.register<ProvideVssDefinitionTask>(PROVIDE_VSS_DEFINITION_TASK) {
+                    val searchPath = extension.searchPath.get().ifEmpty { vssDir }
+                    val vssDefinitionFilePath = StringBuilder(buildDirPath)
+                        .append(fileSeparator)
+                        .append(KSP_INPUT_BUILD_DIRECTORY)
+                        .append(fileSeparator)
+                        .toString()
+                    val vssDefinitionBuildFile = File(vssDefinitionFilePath)
+
+                    logger.info("Searching directory: $searchPath for VSS definitions")
+
+                    val searchDir = file(searchPath)
+                    if (!searchDir.exists()) {
+                        throw FileNotFoundException("Directory for VSS files not found!")
+                    }
+
+                    inputDir = searchDir
+                    outputDir = vssDefinitionBuildFile
+                }
+
+            tasks.getByName("preBuild").finalizedBy(
+                provideVssDefinitionTask.get(),
+            )
+        }
+    }
+
+    companion object {
+        private const val KSP_INPUT_BUILD_DIRECTORY = "kspInput"
+        private const val EXTENSION_NAME = "vssProcessor"
+        private const val PROVIDE_VSS_DEFINITION_TASK = "provideVssDefinition"
+        private const val VSS_FOLDER_NAME = "vss"
+    }
+}
+
+/**
+ * This task takes an input directory [inputDir] which should contain all available VSS definition files and an
+ * output directory [outputDir] where all files are copied to so the VSSProcessor can work with them.
+ */
+private abstract class ProvideVssDefinitionTask : DefaultTask() {
+    @get:Incremental
+    @get:PathSensitive(PathSensitivity.NAME_ONLY)
+    @get:InputDirectory
+    abstract val inputDir: DirectoryProperty
+
+    @get:OutputDirectory
+    abstract val outputDir: DirectoryProperty
+
+    @TaskAction
+    fun provideFile(inputChanges: InputChanges) {
+        inputChanges.getFileChanges(inputDir).forEach { change ->
+            if (change.fileType == FileType.DIRECTORY) return@forEach
+
+            val targetFile = outputDir.file(change.normalizedPath).get().asFile
+            logger.info("Found vss file change for: ${targetFile.name}, change: ${change.changeType}")
```

Comment:
Added a filter of valid extension: "yml, yaml". it will now throw a warning to remove the file:

```
 if (!validVssExtension.contains(extension)) {
                logger.warn("Found incompatible file VSS file: ${file.name} - Consider removing it")
                return@forEach
            }
```